### PR TITLE
Sd 585 add/remove promo code from cart and checkout confirmation

### PR DESCRIPTION
**Pull request by pmilewski**

### CODE REVIEW

**Comment by pmilewski** [comment]
> We should remove this partial from `spree-designs` because it overrides Your changes: https://github.com/spark-solutions/spree-designs/blob/master/app/views/spree/orders/_coupon_code.html.erb


https://github.com/spark-solutions/spree-designs/pull/259

File: frontend/app/views/spree/checkout/edit.html.erb [hunk 1]
```diff
@@ -18,7 +19,12 @@
         <%= form.hidden_field :state_lock_version %>
         <%= render @order.state, form: form %>
       </div>
-      <div id="checkout-summary" class="col-md-12 col-lg-auto" data-hook="checkout_summary_box">
+
+      <% if display_coupon_code %>
+        <%= render 'spree/orders/coupon_code', promotion: @order.adjustments.eligible.first, order_form: form, classes: 'col-lg-4 my-4' %>
```
Comment (damianlegawiec) [inline frontend/app/views/spree/checkout/edit.html.erb]:
```suggestion
        <%= render 'spree/orders/coupon_code', promotion: @order.promotions.coupons.first, order_form: form, classes: 'col-lg-4 my-4' %>
```

File: frontend/app/views/spree/orders/edit.html.erb [hunk 2]
```diff
@@ -36,7 +36,7 @@
             </div>
 
             <% if Spree::Frontend::Config[:coupon_codes_enabled] %>
-              <%= render 'coupon_code', promotion: @order.adjustments.eligible.first, order_form: order_form %>
+              <%= render 'coupon_code', promotion: @order.adjustments.eligible.first, order_form: order_form, classes: '' %>
```
Comment (damianlegawiec) [inline frontend/app/views/spree/orders/edit.html.erb]:
```suggestion
              <%= render 'coupon_code', promotion: @order.promotions.coupons.first, order_form: order_form, classes: '' %>
```

File: frontend/app/views/spree/orders/_coupon_code.html.erb [hunk 3]
```diff
@@ -8,14 +8,15 @@
     </div>
   </div>
 <% else %>
-  <div class='shopping-cart-coupon-code' data-hook='coupon_code'>
+  <div class='shopping-cart-coupon-code <%= classes %>' data-hook='coupon_code'>
     <div class="input-group">
-      <%= order_form.text_field :applied_coupon_code, name: '', size: '30', class: 'form-control border-right-0 disabled', value: promotion.label %>
+      <%= order_form.text_field :applied_coupon_code, name: '', size: '30', class: 'form-control border-right-0 disabled',
+        value: promotion.label,
+        data: { code: promotion.source.promotion.code }
```
Comment (damianlegawiec) [inline frontend/app/views/spree/orders/_coupon_code.html.erb]:
```suggestion
        data: { code: promotion.code }
```

**Comment by pmilewski** [comment]
https://github.com/spark-solutions/spree/pull/145#discussion_r355491484
https://github.com/spark-solutions/spree/pull/145#discussion_r355491823
https://github.com/spark-solutions/spree/pull/145#discussion_r355942281

Unfortunately, the last suggestions which I applied break the tests automatic_promotion_adjustments_spec.rb and spec/features/page_promotions_spec.rb

Could I rollback these changes?

**Pull request merged**